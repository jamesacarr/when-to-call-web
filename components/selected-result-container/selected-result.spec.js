import React from 'react';
import { shallow } from 'enzyme';
import SelectedResult from './selected-result';

describe('SelectedResult', () => {
  const classes = {
    card: 'card',
  };

  it('renders nothing when no result', () => {
    const wrapper = shallow(<SelectedResult classes={classes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const selected = {
      name: 'Testing',
      address: '123 Fake St, Fakeville',
      openingHours: [
        { close: { day: 1, time: '1700' }, open: { day: 1, time: '0900' } },
        { close: { day: 2, time: '1600' }, open: { day: 2, time: '1000' } },
      ],
      utcOffset: -240,
      phone: '1234 5678',
    };

    const wrapper = shallow(<SelectedResult classes={classes} selected={selected} />);
    expect(wrapper).toMatchSnapshot();
  });
});
