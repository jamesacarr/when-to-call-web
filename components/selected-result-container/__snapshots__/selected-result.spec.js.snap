// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedResult renders correctly 1`] = `
<WithStyles(Card)
  className="card"
>
  <WithStyles(CardHeader)
    subheader="123 Fake St, Fakeville"
    title="Testing"
  />
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      gutterBottom={true}
      variant="body1"
    >
      1234 5678
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="body2"
    >
      Call Between
    </WithStyles(Typography)>
    <WithStyles(Table)>
      <WithStyles(TableBody)>
        <WithStyles(OpeningHours)
          data={
            Object {
              "close": Object {
                "day": 1,
                "time": "1700",
              },
              "open": Object {
                "day": 1,
                "time": "0900",
              },
            }
          }
          key="0"
          offset={-240}
        />
        <WithStyles(OpeningHours)
          data={
            Object {
              "close": Object {
                "day": 2,
                "time": "1600",
              },
              "open": Object {
                "day": 2,
                "time": "1000",
              },
            }
          }
          key="1"
          offset={-240}
        />
      </WithStyles(TableBody)>
    </WithStyles(Table)>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;

exports[`SelectedResult renders nothing when no result 1`] = `""`;
